process/dns_v2.go

```diff
@@ -330,9 +330,9 @@ func getDNSNameAsByteSliceByOffset(buf []byte, offset int) (stringasbyteslice []
 	if offset+int(namelen) > len(buf) {
 		return nil, fmt.Errorf("offset out of range [%d:%d] > %d", offset, offset+int(namelen), len(buf))
 	}
-
+	// For when the domain name is an empty string (namelen = 0)
 	if offset+int(namelen) <= offset {
-		return nil, fmt.Errorf("illegal namelen")
+		return nil, fmt.Errorf("illegal domain namelen of 0")
```

Comment:
If this only checks that `namelen` is zero then why isn't it `if namelen == 0` ?
I'm wondering if it's not rather checking for overflow (`namelen` is uint64), if so you might want to check for zero and overflow separately.

Reply:
Thanks for taking a look at this. That seems to be the intent so I added a separate conditional for each check 👍